<launch>

 <!-- joy node -->
  <node respawn="true" pkg="joy"
        type="joy_node" name="xbox_joy" >
    <param name="dev" type="string" value="/dev/input/js0" />
    <param name="deadzone" value="0.1" />
    <param name="autorepeat_rate" value="20" />
    <remap from="/joy" to="/arm_joy" />
  </node>

 <!-- Axes -->
  <param name="SlewAxis" value="0" type="int"/>
  <param name="SlewDir" value="1" type="int"/>
  <param name="ShoulderAxis" value="1" type="int"/>
  <param name="ShoulderDir" value="1" type="int"/>
  <param name="ElbowAxis" value="4" type="int"/>
  <param name="ElbowDir" value="-1" type="int"/>
  <param name="JawRotateAxis" value="3" type="int"/>
  <param name="JawCloseAxis" value="5" type="int"/>
  <param name="WristDir" value="-1" type="int"/>
  <param name="JawOpenAxis" value="2" type="int"/>
  <param name="JawDir" value="1" type="int"/>
  <param name="scale" value="500" type="double"/>
  
  <param name="VehicleModeButton" value="0" type="int"/>  
  <param name="JointsModeButton" value="1" type="int"/>  
  <param name="CartesianModeButton" value="2" type="int"/>  
    
  <param name="ArmBaseFrameId" value="/armbase" type="string"/>  
    
 
 
	<node name="hybrid_teleop_with_dynamics" pkg="joy_controls" type="hybrid_teleop_with_dynamics" output="screen" respawn="false"/>


	<include file="$(find underwater_vehicle_dynamics)/launch/dynamics.launch">
	   <arg name="namespace" value="g500" />
	   <arg name="params" value="$(find underwater_vehicle_dynamics)/config/dynamics_g500.yaml" />
	</include>
	<rosparam param="/g500/dynamics/external_force_topic">g500/ForceSensor</rosparam>
	   
	<arg name="scene" default="shipwreck_circuito.xml" />
	<node name="uwsim" pkg="uwsim" type="uwsim" args="--configfile $(arg scene)" output="screen" respawn="false" />
	<!-- UNSUED RIGTH NW arg name="thrusters_topic" default="/g500/thrusters_input" / -->
	
	<!-- node name="keyboard_command" pkg="underwater_vehicle_dynamics" type="keyboard.py" args="$(arg thrusters_topic)" output="screen" respawn="false"/ -->

</launch>

